sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -190,6 +190,19 @@ trait V2CreateTablePlan extends LogicalPlan {
   def withPartitioning(rewritten: Seq[Transform]): V2CreateTablePlan
 }
 
+/** A trait used for logical plan nodes that create or replace V2 table definitions. */
+trait V2CreateTablePlanX extends LogicalPlan {
```

Comment:
@huaxingao  It is from your https://github.com/apache/spark/pull/34060, I will rebase and change this PR to ready when it is merged